### PR TITLE
implemented a pyth sdk

### PR DESCRIPTION
**Pull request by chlenc**

### CODE REVIEW

**Comment by K1-R1** [comment]
@chlenc Have you signed the CLA agreement above? It still says it hasn't been signed for me

File: src/constants.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+pub const RPC: &str = "beta-5.fuel.network";
+
+pub const TOKEN_CONTRACT_ID: &str =
```
Comment (K1-R1) [inline src/constants.rs:3]: What token is this?

File: src/constants.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "0x3bcfb9d38a65fd39a829e5137ebf4bb2a61429d319a2899e457fce0b436ca1a6";
+pub const USDC_ASSET_ID_STR: &str =
+    "0x0450e4d385cbd2914f74505f18f01587cc4f4ad1fdef4b80cbde2a8155a86d72";
+pub const BTC_ASSET_ID_STR: &str =
+    "0x593b117a05f5ea64b39ba1f9bc3fb7e7a791c9be130e28376ad552eacdb3b746";
+pub const UIN_ASSET_ID_STR: &str =
```
Comment (K1-R1) [inline src/constants.rs:9]: Is this `UIN` or `UNI`?

File: src/pyth_utils.rs [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+    name = "PythOracleContract",
+    abi = "pyth-contract/out/debug/pyth-contract-abi.json"
+));
+
+pub struct Pyth {
+    pub instance: PythOracleContract<WalletUnlocked>,
+    pub wallet: WalletUnlocked,
+}
+
+const ORACLE_CONTRACT_BINARY_PATH: &str = "./pyth-contract/out/debug/pyth-contract.bin";
```
Comment (K1-R1) [inline src/pyth_utils.rs:24]: The remaining constants should be added to the constants file, and test values should be distinguished

File: tests/utils/setup.rs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    .await
+    .unwrap();
+    let deployer_wallet = wallets.pop().unwrap();
+
+    let pyth = Pyth::deploy(&deployer_wallet).await;
+
+    (pyth.instance.contract_id().into(), pyth)
+}
+
+// const ORACLE_CONTRACT_BINARY_PATH: &str = "./out/debug/pyth-contract.bin";
```
Comment (K1-R1) [inline tests/utils/setup.rs:27]: Commented code should be removed if deprecated

**Review by K1-R1 — CHANGES_REQUESTED** [review]
I'll implement the requested changes below

File: src/constants.rs (hunk 1 above)
Comment (K1-R1) [reply src/constants.rs:3]: @chlenc

File: src/constants.rs (hunk 2 above)
Comment (K1-R1) [reply src/constants.rs:9]: @chlenc

File: src/constants.rs (hunk 1 above)
Comment (chlenc) [reply src/constants.rs:3]: It’s our contract based on fuel token standards, not sure it need to be here

Comment (chlenc) [reply src/constants.rs:3]: https://github.com/compolabs/src-20

File: src/constants.rs (hunk 2 above)
Comment (chlenc) [reply src/constants.rs:9]: This is UNI token that was issued used our token contract 
https://github.com/compolabs/src-20

File: src/pyth_utils.rs (hunk 3 above)
Comment (chlenc) [reply src/pyth_utils.rs:24]: 100%

File: tests/utils/setup.rs (hunk 4 above)
Comment (chlenc) [reply tests/utils/setup.rs:27]: Agree

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**